lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,238 @@
+    state: absent
+    origin_access_identity_id: EBXCCWOVSAYYD
+
+'''
+
+RETURN = '''
+
+location:
+    description: describes a url specifying the origin access identity.
+    returned: always
+    type: str
+
+'''
+
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import boto3_conn, HAS_BOTO3
+from ansible.modules.cloud.amazon.cloudfront_distribution import CloudFrontHelpers
+from ansible.modules.cloud.amazon.cloudfront_facts import CloudFrontFactsServiceManager
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.basic import AnsibleModule
+from botocore.signers import CloudFrontSigner
+import datetime
+from functools import partial
+import json
+import traceback
+
+try:
+    import botocore
+except ImportError:
+    pass
+
+
+class CloudFrontOriginAccessIdentityServiceManager:
+    """
+    handles cloudfront origin access identity service calls to aws
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.create_client('cloudfront')
+
+    def create_client(self, resource):
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(self.module, boto3=True)
+            self.client = boto3_conn(self.module, conn_type='client', resource=resource,
+                                     region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except botocore.exceptions.NoRegionError:
+            self.module.fail_json(msg=("region must be specified as a parameter in "
+                                       "AWS_DEFAULT_REGION environment variable or in "
+                                       "boto configuration file"))
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(msg="unable to establish connection - " + str(e),
+                                  exception=traceback.format_exc(),
+                                  **camel_dict_to_snake_dict(e.response))
+
+    def create_origin_access_identity(self, caller_reference, comment):
+        try:
+            func = partial(self.client.create_cloud_front_origin_access_identity,
+                           CloudFrontOriginAccessIdentityConfig={
+                               'CallerReference': caller_reference,
+                               'Comment': comment
+                           })
+            return self.paginated_response(func)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(msg="error creating cloud front origin access identity - " + str(e),
+                                  exception=traceback.format_exc(),
+                                  **camel_dict_to_snake_dict(e.response))
+
+    def delete_origin_access_identity(self, origin_access_identity_id, e_tag):
+        try:
+            func = partial(self.client.delete_cloud_front_origin_access_identity,
+                           Id=origin_access_identity_id, IfMatch=e_tag)
```

Comment:
This isn't ever going to be paginated. [docs](http://boto3.readthedocs.io/en/latest/reference/services/cloudfront.html#CloudFront.Client.create_cloud_front_origin_access_identity)